src/model/Model_CustomField.cpp

```diff
@@ -0,0 +1,193 @@
+{
+    std::make_pair(Model_CustomField::STRING, wxString("String")),
+    std::make_pair(Model_CustomField::INTEGER, wxString("Integer")),
+    std::make_pair(Model_CustomField::DECIMAL, wxString("Decimal")),
+    std::make_pair(Model_CustomField::BOOLEAN, wxString("Boolean")),
+    std::make_pair(Model_CustomField::DATE, wxString("Date")),
+    std::make_pair(Model_CustomField::TIME, wxString("Time")),
+    std::make_pair(Model_CustomField::SINGLECHOICHE, wxString("SingleChoiche")),
+    std::make_pair(Model_CustomField::MULTICHOICHE, wxString("MultiChoiche"))
+};
+
+Model_CustomField::Model_CustomField()
+: Model<DB_Table_CUSTOMFIELD_V1>()
+{
+}
+
+Model_CustomField::~Model_CustomField()
+{
+}
+
+/**
+* Initialize the global Model_CustomField table.
+* Reset the Model_CustomField table or create the table if it does not exist.
+*/
+Model_CustomField& Model_CustomField::instance(wxSQLite3Database* db)
+{
+    Model_CustomField& ins = Singleton<Model_CustomField>::instance();
+    ins.db_ = db;
+    ins.destroy_cache();
+    ins.ensure(db);
+
+    return ins;
+}
+
+/** Return the static instance of Model_CustomField table */
+Model_CustomField& Model_CustomField::instance()
+{
+    return Singleton<Model_CustomField>::instance();
+}
+
+///** Return a dataset with fields linked to a specific object */
+//const Model_CustomField::Data_Set Model_CustomField::GetFields(Model_Attachment::REFTYPE RefType)
+//{
+//    Data_Set fields;
+//    wxString reftype_desc = Model_Attachment::reftype_desc(RefType);
+//    for (const auto & field : this->find(Model_CustomField::DB_Table_CUSTOMFIELD_V1::REFTYPE(RefType)))
+//    {
+//        fields.push_back(field);
+//    }
+//    return field;
+//}
+
+/** Delete a field and all his data */
+bool Model_CustomField::Delete(const int& FieldID)
+{
+    this->Savepoint();
+    for (const auto& r : Model_CustomFieldData::instance().find(Model_CustomFieldData::FIELDID(FieldID)))
+        Model_CustomFieldData::instance().remove(r.id());
+    this->ReleaseSavepoint();
+    return this->remove(FieldID, db_);
+}
+
+wxString Model_CustomField::fieldtype_desc(const int FieldTypeEnum)
+{
+    const auto& item = FIELDTYPE_CHOICES[FieldTypeEnum];
+    wxString reftype_desc = item.second;
+    return reftype_desc;
+}
+
+Model_CustomField::FIELDTYPE Model_CustomField::type(const Data* r)
+{
+    for (const auto& item : FIELDTYPE_CHOICES) if (item.second.CmpNoCase(r->TYPE) == 0) return item.first;
+
+    return FIELDTYPE(-1);
+}
+
+Model_CustomField::FIELDTYPE Model_CustomField::type(const Data& r)
+{
+    return type(&r);
+}
+
+wxArrayString Model_CustomField::all_type()
+{
+    wxArrayString types;
+    for (const auto& item : FIELDTYPE_CHOICES) types.Add(item.second);
+    return types;
+}
+
+wxString Model_CustomField::getTooltip(const wxString& Properties)
+{
+    json::Object jsonProperties;
+    std::wstringstream jsonPropertiesStream;
+
+    jsonPropertiesStream << Properties.ToStdWstring();
```

Comment:
To start working with this branch I've thanged it to :
~~~~~
jsonPropertiesStream << (Properties.empty() ? L"{}" : Properties.ToStdWstring());
~~~~~
for any jsonPropertiesStream in this file.
In other case the Error message appears: Unexpected end o ftocken string.